src/document_models.py

```diff
@@ -32,8 +46,111 @@ class Person(Document):
     preferredName = StringField(max_length=240)
     services = DictField()
 
-    # TODO:
+    # TODO: other fields
     # add role at Olin
     # BOW students?
     # allergies/diet
-    # image/gravatar
\ No newline at end of file
+    # image/gravatar
+
+class Token(Document):
+    """ Represents an authorization token that allows a user to delegate a client
+    application access to the Olin API on their behalf. """
+    email = EmailField(max_length=100, required=True, unique=True)
+    # 'validated' is marked as true once the user has confirmed they own the
+    # email address the validation token is sent to
+    validated = BooleanField(default=False)
+
+    def generate_token(self, expiration=86400 * 365):
+        """ Generate an authentication token, by default good for 1 year """
+        serializer = Serializer(
+            current_app.config['SECRET_KEY'], expires_in=expiration)
+        return serializer.dumps({'email': self.email}).decode(encoding='UTF-8')
```

Comment:
It's a cryptographically secure token - docs here: http://pythonhosted.org/itsdangerous/#json-web-signatures

(basically it does encoding, yes, but the encoding is more importantly a vector for ensuring that the token is valid using crypto)